draft-ietf-quic-transport.md

```diff
@@ -1489,7 +1489,9 @@ Differences in routing on the Internet might cause the same destination address
 and connection ID to reach a different server instance which does not posses the
 necessary connection state. Receiving a Stateless Reset in response to a probing
 packet SHOULD NOT terminate the connection, but MUST cause the endpoint to
-consider path validation to have failed.
+consider path validation to have failed.  After receiving a Stateless Reset, the
+client MUST NOT send additional packets with the Connection ID used on the
+probing packet.
 
```

Comment:
with high probability.

Reply:
With high probability what?  This isn't a statement that you probably reached the wrong server, it's a normative requirement that the client abandon the path if a certain response occurs.

Reply:
When you create new connections, you use a random connection ID. If "high probability" is not there, it means that you have to log all previous connections and test against those, otherwise you can just assume the random ID is unlikely to collide, but if it does, the world will continue to exist as best as it can.

Reply:
I'm not following.  This text isn't about new connections, and the CID isn't random -- it's supplied by the server.

Reply:
If you create a new Initial packet for a new connection, you use a random CID before using the server provided CID. If you consider this a separate namespace, then your are right. And if you are only referring to the current connection you are also right. But otherwise you risk accidentially creating an already used ID in a new connection, and with, say, 32 bit CID's it is non-neglectible.

Reply:
Given that this text is about path validation using probing packets, yes, I'm assuming we're only talking about the current connection.

Reply:
Ah, I think I see what you're saying now.  If the prohibition is read as "you must never use this CID again for anything ever," then the client would be obligated to keep such CIDs persisted and make sure that it never has a randomly-chosen initial CID on a future connection which happens to collide with a CID that got a stateless reset at some point on a past connection.  Yeah, this prohibition is connection-scoped, because now the Stateless Reset token has been leaked and it's vulnerable to packet injection attacks.

Reply:
thanks